### DISCUSSION TITLE
How to monitor same application deployed by different methods

### DISCUSSION THREAD

**Discussion by mateusoliveira43** [dora-metrics/pelorus#895] [General]
How should Pelorus monitor an application that is deployed, for example, to 3 different namespaces?

This discussion can go even further: How should Pelorus monitor an application that is deployed to different scenarios, outside a cluster (without namespace) or without creating an container image (associated image sha)?

**Comment by mateusoliveira43** [comment]
One problem, as we discussed in meeting, is how to associate each data to each application. Should we first register an application, and only gather data about it if it is already registered?

Should we try to validate data before sending to Prometheus (like checking if commit_hash already exists to a given application) (if so, when/how this validation should occur?)

Do we need to collect more data (to identify an application related to it)?

Do we need to collect less data? Namespace seems to be unnecessary, since we have support to collecting data outside a cluster, and a commit/deploy will always create a container image related to it?

**Comment by mpryc** [comment]
How about using label with a `/` separated schema:

examples:
```
app.kubernetes.io/name=organization/group/myapp
app.kubernetes.io/name=organization/anothergroup/myapp
app.kubernetes.io/name=anotherorganization/myapp

app.kubernetes.io/name=myapp
```

The last one is to provide compatibility with the simple scenarios as in the current default one, but the top three allows to have many org/group/subgroup/other... parts to identify a single app instance ?
